### ISSUE TITLE
riot-desktop crashes when attempting to attach a file on KDE system

### ISSUE THREAD

**Issue by hjones2199** [NixOS/nixpkgs#83544]
**Describe the bug**
The riot-desktop package crashes on NixOS systems where KDE is the only explicitly installed
desktop environment.

**To Reproduce**
1. Install KDE and riot-desktop
2. launch riot-desktop, sign in and join a room
3. click the paperclip to attach a file

**Expected behavior**
The KDE file chooser should pop up allowing you to select a file

**Additional context**
My best guess based on the terminal output is that Matrix is trying to use some kind of GTK library to determine the file chooser and does not list it as a dependency.

Terminal output of the crashing program:
> Seshat isn't installed, event indexing is disabled.
> xdg-mime: application argument missing
> Try 'xdg-mime --help' for more information.
> Fontconfig warning: "/etc/fonts/fonts.conf", line 86: unknown element "blank"
> No update_base_url is defined: auto update is disabled
> (node:20600) ProtocolDeprecateCallback: The callback argument of protocol module APIs is no 
> longer needed.
> (node:20600) ProtocolDeprecateCallback: The callback argument of protocol module APIs is no 
> longer needed.
> [20661:0327/182117.679930:ERROR:buffer_manager.cc(488)] [.DisplayCompositor]GL ERROR 
> :GL_INVALID_OPERATION : glBufferData: <- error from previous GL command

> (electron:20600): GLib-GIO-ERROR **: 18:21:28.629: No GSettings schemas are installed on the 
> system
> [1]    20600 trace trap (core dumped)  riot-desktop

**Comment by rnhmjoj** [comment]
This is an electron program, it probably needs to be wrapped manually, the standard hook won't work.

**Comment by rnhmjoj** [comment]
Backported to 20.03 in ef955ecb1ea10aaa95d7221048e1f30180929116.

**Comment by tasmo** [comment]
Now riot-desktop crashes here with the following error:

```
(electron:25597): GLib-GIO-ERROR **: 18:09:09.944: Settings schema 'org.gtk.Settings.FileChooser' is not installed
[1]    25597 trace trap (core dumped)  riot-desktop
```

**Comment by rnhmjoj** [comment]
@tasmo: there has been a new PR that fixes up this one: #84285. Can you check that out?

**Comment by tasmo** [comment]
@rnhmjoj Yes, I'm on master with the latest commits. The `riot-desktop.nix` shows at the top:

```nix
  { stdenv, fetchFromGitHub
  , makeWrapper, makeDesktopItem, mkYarnPackage
  , electron_7, riot-web, gtk3
  , wrapGAppsHook, glib
  }:
```
